network/src/network.rs

```diff
@@ -579,18 +582,12 @@ impl ServiceHandle for EventHandler {
                 debug!("ProtocolError({}, {}) {}", id, proto_id, error);
                 let message = format!("ProtocolError id={}", proto_id);
                 // Ban because misbehave of remote peer
-                if let P2pError::IoError(_) = error {
-                    self.network_state.ban_session(
-                        &context.control(),
-                        id,
-                        Duration::from_secs(300),
-                        message,
-                    );
-                } else if let Err(err) =
-                    disconnect_with_message(context.control(), id, message.as_str())
-                {
-                    debug!("Disconnect failed {:?}, error {:?}", id, err);
-                }
+                self.network_state.ban_session(
```

Comment:
It's hard to tell why is `if else` is removed in this PR. Could you please give more context on this?

Reply:
tentacle splits the error, the output is more detailed, only one error will be output here, which is previous `if let` error.

This is the reason why a big Error output is not used